src-ui/lib/openLibrary.ts

```diff
@@ -1,12 +1,35 @@
-import type { paths, components } from 'open-library-api'
-import { type Book } from './types/book.js'
+import type { components, paths } from 'open-library-api'
 import createClient from 'openapi-fetch'
+import { type BookWithoutId } from './types/book.js'
 
-const client = createClient<paths>({ baseUrl: 'https://openlibrary.org/' })
+const fetchWithTimeout = async (request: Request | string, timeout = 3000): Promise<Response> => {
+  const controller = new AbortController()
+  const timeoutId = setTimeout(() => {
+    controller.abort()
+  }, timeout)
 
-export async function getByISBN(isbn: string): Promise<Book> {
+  return await fetch(request, { signal: controller.signal })
+    .then((response) => {
+      clearTimeout(timeoutId)
+      return response
+    })
+    .catch((error) => {
+      clearTimeout(timeoutId)
+      if (error.name === 'AbortError') {
+        throw new Error('Fetch request timed out')
```

Comment:
Over time, we could add other ways to abort a request. Imagine a user navigates away from the page, for example. A more accurate long-term description would be `'Fetch request aborted (perhaps due to a timeout)'`.